### PR TITLE
Partly revert removal of docs and coverage CI

### LINKED ISSUE #1689: Fix Zebra's failing docs CI
**Is your feature request related to a problem? Please describe.**

Zebra's docs CI is consistently failing.

For example:
https://github.com/ZcashFoundation/zebra/commit/bd03a51482a3b2e4f7997b234370dbb286d0aa75
https://github.com/ZcashFoundation/zebra/runs/1826774050

**Describe the solution you'd like**

We disabled the docs CI temporarily in #1688.

- [ ] investigate the cause of the failures
- [ ] fix the failures
- [ ] re-enable the docs CI

**Additional context**

The coverage CI is also failing.

### LINKED ISSUE #1690: Fix Zebra's failing coverage CI
**Is your feature request related to a problem? Please describe.**

Zebra's coverage CI is consistently failing.

For example:
https://github.com/ZcashFoundation/zebra/commit/bd03a51482a3b2e4f7997b234370dbb286d0aa75
https://github.com/ZcashFoundation/zebra/runs/1826774021

**Describe the solution you'd like**

We disabled the coverage CI temporarily in #1688.

- [ ] investigate the cause of the failures
- [ ] fix the failures
- [ ] re-enable the coverage CI
- [ ] name it explicitly 'nightly' 
- [ ] Check that we're not exceeding the 20 concurrent jobs limit: https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

**Additional context**

The docs CI is also failing.

### PR DESCRIPTION
**Pull request by oxarbitrage**
## Motivation

Research and fix the cause of failing CI docs and coverage.

## Solution

Revert https://github.com/ZcashFoundation/zebra/pull/1688 fully but use `beta` as the toolchain instead of `nightly` in the docs. For the coverage we need to stay in nightly as coverage uses a compiler option `-Z` only available there. 

More info about docs failure at https://github.com/Alexhuszagh/rust-lexical/issues/55

We used nightly because of some nightly-only docs feature that are available in beta now.

## Review

@dconnolly after CI pass.

## Related Issues

Close https://github.com/ZcashFoundation/zebra/issues/1689  and close https://github.com/ZcashFoundation/zebra/issues/1690

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+jobs:
+
+  coverage:
+    name: Coverage
```
Comment (dconnolly) [inline .github/workflows/coverage.yml]:
```suggestion
    name: Coverage (+nightly)
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+
+  build:
+    name: Build and Deploy Docs
```
Comment (dconnolly) [inline .github/workflows/docs.yml]:
```suggestion
    name: Build and Deploy Docs (+beta)
```

**Comment by teor2345** [comment]
This is a partial revert, because some of the reverted config has been changed to beta rust

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  push:
+    branches:
+      - main
+
+jobs:
+
+  coverage:
+    name: Coverage (+nightly)
+    timeout-minutes: 30
+    runs-on: ubuntu-20.04
```
Comment (dconnolly) [inline .github/workflows/coverage.yml]: Should we leave this on `-latest`?

Comment (oxarbitrage) [reply .github/workflows/coverage.yml]: We can try, i am not sure if we are using ubuntu 20 here for some reason.

I saw some warnings in actions about this so all latest will be in ubuntu-20 soon:

> Ubuntu-latest workflows will use Ubuntu-20.04 soon. For more details, see https://github.com/actions/virtual-environments/issues/1816

Comment (oxarbitrage) [reply .github/workflows/coverage.yml]: Trying with https://github.com/ZcashFoundation/zebra/pull/1758/commits/5edc34d91c7ee2204c4a75c331934e0669e227ab

Comment (dconnolly) [reply .github/workflows/coverage.yml]: Sure but we don't want to pin on -20.04, we want to keep up with -latest on each platform

Comment (oxarbitrage) [reply .github/workflows/coverage.yml]: Definitely agree

**Review by dconnolly — APPROVED** [review]

**Pull request merged**